rpy2/ipython/rmagic.py

```diff
@@ -169,6 +170,50 @@ def py2rpy_list(obj):
     return res_rpy
 
 
+def _find(name: str, ns: dict):
+    """Find a Python name, which might include dot-separated path to a name.
+
+    Args:
+    - name: a key in dict if no dot ('.') in it, otherwise
+    and sequence of dot-separated namespaces with the name of the
+    object last (e.g., `package.module.name`).
+    Returns:
+    The object wanted. Raises a NameError or an Attribute error if not found.
+    """
+
+    obj = None
+    obj_path = name.split('.')
+    look_for_i = 0
+    try:
+        obj = ns[obj_path[look_for_i]]
+    except KeyError:
+        raise NameError(
+            f"name '{obj_path[look_for_i]}' is not defined."
+        )
```

Comment:
```suggestion
    except KeyError as e:
        message = f"name '{obj_path[look_for_i]}' is not defined."
        if obj_path[look_for_i] == "":
            message += ' Did you forget to remove trailing comma `,` or included spaces?'
        raise NameError(message) from e
```


for nicer traceback and less user frustration?

Reply:
Makes sense. Thanks.